### DISCUSSION TITLE
How to appropriately upsert and connect

### DISCUSSION THREAD

**Discussion by powderham** [prisma/prisma#2347] [Q&A] [answered]
I have a model `Event` to which I want to connect an `Artist` when I am creating the `Event` document.

The Artist can be previously defined, so the input current can be either 

```
{
 Artist: [{name: 'foo'}] or [{name: 'bar', id: 1}]
}
```

I believe I need to upsert the Artists first so I run the following code: 

```
const artists = await Promise.all(
          Artist.map((artist) =>
            prisma.artist.upsert({
              create: omit(artist, "id"),
              update: artist,
              where: artist,
            })
          )
        );

// omit imported from lodash
```

It feels like an anti pattern to have to restrict the id.

My question is, is there a way I can nested upsert Artists in the Event creation, or handle the existence of the artist id in the create more elegantly?

**Comment by ryands17** [comment, marked as answer]
Hey @powderham :wave:
 
Unfortunately, the current workaround for this would be omitting the `id` as the query wouldn't work without that. So it would be either as you have done above or via destructuring the `Artist`.

Could you pls open a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) including the description above so that we can look into this?

**Comment by ludralph** [comment]
Hi @powderham,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
